docs/design/fabric-v2+/fpc-lifecycle-v2.puml

```diff
@@ -15,124 +15,149 @@ title Foot Box removed
-    FPC is expected to support multiple endorsers in the future. To have an idea of how FPC will
-    enalve multiple endorsers, please check out the diagram describing
-    how secret/private keys are expected to be distributed among endorsing chaincode enclaves.
+    Sequence: sequence_number,
+    Endorsement Policy: (2of3),
+    Plugins: ecc-vscc (FPC validation plugin)
+
+    NOTE: the admin approves the chaincode definition
+    including the **mrenclave** that was computed during
+    code inspection.
+
+    NOTE: the endorsement policy specifies how many
+    enclaves are expected to run the chaincode. The MVP
+    implementation will support a "designated" endorsing
+    peer only.
+    Hence, initially, the endorsement policy will require
+    a single endorser. FPC is expected to support
+    multiple endorsers in the future. To have an
+    idea of how FPC will enclave multiple endorsers,
+    please check out key distribution diagram.
+
     NOTE: FPC does not support custom user plugins.
-end note
-Peer1_CLI -> Orderer ++: approve transaction
-return transaction committed /' Peer1_CLI -> Orderer '/
-return /' FPC_Peer1_CLI_Wrapper -> Peer1_CLI '/
-note over FPC_Peer1_CLI_Wrapper
-    Bookkeping operations.
-    Note: packageId is an input parameter of approveformyorg
-end note
-FPC_Peer1_CLI_Wrapper -> FPC_Peer1_CLI_Wrapper : if packageId was stored (and hence it is an FPC chaincode) then store mapping packageid to <cc_name, cc_version>
-return /' Admin -> FPC_Peer1_CLI_Wrapper '/
-end /' group approve '/
-
-group other orgs approve
-Channel -> Channel : approvals from other parties
-note over Channel
-    Other participants must approve the chaincode definition,
-    and commit the corresponding transaction,
-    according to the LifecycleEndorsement policy.
-    So the party here has to loop to check for the approvals.
-end note
-end group 'other orgs approve'
+    end note
+
+    Peer_CLI -> Orderer++ : approve transaction
+    return transaction committed /' Peer1_CLI -> Orderer '/
+
+    note right Peer_CLI
+    Bookkeeping operations.
+    NOTE: packageId is an input parameter of approveformyorg
+    end note
 
+    Peer_CLI -> Peer_CLI : if packageId was stored (and hence it is an FPC chaincode)\n\tthen store mapping packageId to <cc_name, cc_version>
+    return
+end
 
 loop until enough approvals
-	Admin -> Peer1_CLI : lifecycle chaincode checkcommitreadiness
-note right
+	Admin -> Peer_CLI++ : lifecycle\nchaincode checkcommitreadiness
+    note right
     **Chaincode definition**
     (implemented to match the approved one)
-end note
+    end note
+
+    note right Peer_CLI
+    Other participants must approve the chaincode definition
+    by committing the corresponding transaction according to
+    the LifecycleEndorsement policy.
+
+    Note that other orgs follow the same lifecycle steps as
+    illustrated here, in particular, they inspect the FPC
+    chaincode and generate **mrenclave** accordingly.
```

Comment:
If you really want to emphasize the mrenclave also here i would suggest we emphasize also that it must be the _same_ one (and hence we do not support the n-programming possible in Fabric since v2)
```suggestion
    Note that other orgs follow the same lifecycle steps as
    illustrated here, in particular, they inspect the FPC
    chaincode and must approve the same **mrenclave** (and hence identical code).  Requiring identical code is a deparature of the new capability added to Fabric v2 in allowing different peers to have different implementations. Unfortunately, due to confidentiality requirements, such a feature is not possible for FPC.
```